tests/annotations/tests.py

```diff
@@ -292,7 +307,7 @@ def test_full_expression_annotation_with_aggregation(self):
     def test_aggregate_over_full_expression_annotation(self):
         qs = Book.objects.annotate(
             selected=ExpressionWrapper(~Q(pk__in=[]), output_field=BooleanField()),
-        ).aggregate(Sum("selected"))
+        ).aggregate(selected__sum=Sum(Cast("selected", IntegerField())))
```

Comment:
This test was flawed on databases that have a proper boolean type and only happened to work because of how `BooleanField.select_format` was previously implemented to return an integer instead of a boolean expression

https://github.com/django/django/blob/444b6da7cc229a58a2c476a52e45233001dc7073/django/db/models/fields/__init__.py#L1106-L1113

I assume this was done to appease Oracle lack of handling for selected boolean expression.